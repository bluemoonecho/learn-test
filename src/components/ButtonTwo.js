import React from 'react'

const ButtonTwo = ({label}) => {
    return (
        <div style={{
            margin: '50px auto',
            border: '1px solid gray',
            padding: '10px',
            width: '150px'
        }}
        data-testid='button'>
            Hi {label}
        </div>
    )
}

export default ButtonTwo
