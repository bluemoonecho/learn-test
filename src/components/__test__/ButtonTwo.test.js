import React from 'react'
import ButtonTwo from './../ButtonTwo'
import {render, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup);
it('renders without crashing', () =>{
    const div = document.createElement('div')
    render(<ButtonTwo/>, div)
})

it('renders button correctly', ()=>{
    const {getByTestId} =  render(<ButtonTwo label='save'/>)
    expect(getByTestId('button')).toHaveTextContent('save')
})

it('matches snapshot 1', ()=>{
    const tree = renderer.create(<ButtonTwo label='save'></ButtonTwo>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 2', ()=>{
    const tree = renderer.create(<ButtonTwo label='click me please'></ButtonTwo>).toJSON();
    expect(tree).toMatchSnapshot();
})
