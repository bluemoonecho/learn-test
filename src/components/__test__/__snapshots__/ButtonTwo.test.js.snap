// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  data-testid="button"
  style={
    Object {
      "border": "1px solid gray",
      "margin": "50px auto",
      "padding": "10px",
      "width": "150px",
    }
  }
>
  Hi 
  save
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  data-testid="button"
  style={
    Object {
      "border": "1px solid gray",
      "margin": "50px auto",
      "padding": "10px",
      "width": "150px",
    }
  }
>
  Hi 
  click me please
</div>
`;
